src/main/java/de/srendi/advancedperipherals/common/addons/computercraft/peripheral/RSBridgePeripheral.java

```diff
@@ -0,0 +1,887 @@
+    }
+
+    /**
+     * imports a fluid to the system from a valid tank
+     *
+     * @param arguments  the arguments given by the computer
+     * @param targetTank the give tank
+     * @return the imported amount or null with a string if something went wrong
+     */
+    protected MethodResult importToRS(@NotNull IArguments arguments, @Nullable IChemicalHandler targetTank) throws LuaException {
+        RsChemicalHandler chemicalHandler = new RsChemicalHandler(getNetwork());
+        Pair<ChemicalFilter, String> filter = ChemicalFilter.parse(arguments.getTable(0));
+
+        if (filter.rightPresent())
+            return MethodResult.of(0, filter.getRight());
+
+        if (targetTank == null)
+            return MethodResult.of(0, "Target Tank does not exist");
+
+        return MethodResult.of(ChemicalUtil.moveChemical(targetTank, chemicalHandler, filter.getLeft()), null);
+    }
+
+    @Override
+    @LuaFunction(mainThread = true)
+    public MethodResult isConnected() {
+        return MethodResult.of(isAvailable());
+    }
+
+    @Override
+    @LuaFunction(mainThread = true)
+    public MethodResult isOnline() {
+        if (!isAvailable())
+            return notConnected();
+
+        return MethodResult.of(getComponent(EnergyNetworkComponent.class).getStored() > 0);
+    }
+
+    @Override
+    @LuaFunction(mainThread = true)
+    public MethodResult getItem(IArguments arguments) throws LuaException {
+        if (!isAvailable())
+            return notConnected();
+
+        Pair<ItemFilter, String> filter = ItemFilter.parse(arguments.getTable(0));
+        if (filter.rightPresent())
+            return MethodResult.of(null, filter.getRight());
+
+        ItemFilter parsedFilter = filter.getLeft();
+        if (parsedFilter.isEmpty())
+            return MethodResult.of(null, "EMPTY_FILTER");
+
+        Map<?, ?> resourceProperties = RsApi.getParsedItem(getNetwork(), parsedFilter);
+        if (resourceProperties == null)
+            return MethodResult.of(null, "NOT_FOUND");
+
+        return MethodResult.of(resourceProperties);
+    }
+
+    @Override
+    @LuaFunction(mainThread = true)
+    public MethodResult getFluid(IArguments arguments) throws LuaException {
+        if (!isAvailable())
+            return notConnected();
+
+        Pair<FluidFilter, String> filter = FluidFilter.parse(arguments.getTable(0));
+        if (filter.rightPresent())
+            return MethodResult.of(null, filter.getRight());
+
+        FluidFilter parsedFilter = filter.getLeft();
+        if (parsedFilter.isEmpty())
+            return MethodResult.of(null, "EMPTY_FILTER");
+
+        Map<?, ?> resourceProperties = RsApi.getParsedFluid(getNetwork(), parsedFilter);
+        if (resourceProperties == null)
+            return MethodResult.of(null, "NOT_FOUND");
+
+        return MethodResult.of(resourceProperties);
+    }
+
+    @Override
+    public MethodResult getChemical(IArguments arguments) throws LuaException {
```

Comment:
missing `LuaFunction` I think? also I believe CC:T like you mark every `LuaFunction` with final keyword, since they should never be override.

Reply:
Good catch
There are also another functions with missing annotations and final keywords